### PR TITLE
vm repair changes to support repair for cvm

### PR DESCRIPTION
**Pull request by dkulkarni-ms**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az vm repair create -g RG -n VM --unlock-encrypted-vm --encrypted-vm-recovery-password "***" 
Above command creates confidential rescue VM. 

az vm repair run -g RG -n VM --run-id win-crowdstrike-fix-bootloop-cvm --run-on-repair
This to run script on disk. 

az vm repair restore -g RG -n VM  --verbose --debug> debugoutput

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Review by canfikret — CHANGES_REQUESTED** [review]

File: src/vm-repair/azext_vm_repair/repair_utils.py [hunk 1]
```diff
@@ -462,15 +462,22 @@ def _unlock_mount_linux_encrypted_disk(repair_vm_name, repair_group_name):
...
 
-def _unlock_mount_windows_encrypted_disk(repair_vm_name, repair_group_name):
+def _unlock_mount_windows_encrypted_disk(repair_vm_name, repair_group_name, encrypted_vm_recovery_password):
     # Unlocks the disk using the phasephrase and mounts it on the repair VM.
+    if encrypted_vm_recovery_password:
+        logger.info('Using bitlocker password to unlock...')
+        WINDOWS_RUN_SCRIPT_NAME = 'win-mount-encrypted-disk-bitlockerV.ps1'
+        BITLOCKER_RECOVERY_PARAMS = []
+        BITLOCKER_RECOVERY_PARAMS.append('bitlockerkey="{}"'.format(encrypted_vm_recovery_password))
+        return _invoke_run_command(WINDOWS_RUN_SCRIPT_NAME, repair_vm_name, repair_group_name, False, parameters=BITLOCKER_RECOVERY_PARAMS)
```
[the remark this answers is not in the document]
Comment (dkulkarni-ms) [reply src/vm-repair/azext_vm_repair/repair_utils.py:472]: I had tested this scenario of wrong recovery password, the entire script fails and writes error. May be I will raise exception if script fails and fail entire create command.

File: src/vm-repair/azext_vm_repair/_params.py [hunk 2]
```diff
@@ -29,6 +29,7 @@ def load_arguments(self, _):
         c.argument('copy_disk_name', help='Name of OS disk copy.')
         c.argument('repair_group_name', help='Name for new or existing resource group that will contain repair VM.')
         c.argument('unlock_encrypted_vm', help='Option to auto-unlock encrypted VMs using current subscription auth.')
+        c.argument('encrypted_vm_recovery_password', help='Option to auto-unlock encrypted VMs using current subscription auth.')
```
[the remark this answers is not in the document]
Comment (dkulkarni-ms) [reply src/vm-repair/azext_vm_repair/_params.py:32]: Changing

File: src/vm-repair/azext_vm_repair/repair_utils.py [hunk 3]
```diff
@@ -490,6 +497,26 @@ def _fetch_compatible_windows_os_urn(source_vm):
...
+    else:
+        # If source SKU not available then defaulting 2022 datacenter image. Also narrowing on gen 2 as CVMs require Gen2.
+        fetch_urn_command = 'az vm image list -s "2022-Datacenter" -f WindowsServer -p MicrosoftWindowsServer -l {loc} --verbose --all --query "[?sku==\'2022-datacenter-core-smalldisk-g2\'].urn | reverse(sort(@))" -o json'.format(loc=location)
+
+    logger.info('Fetching compatible Windows OS images from gallery...')
+    urns = loads(_call_az_command(fetch_urn_command))
+
+    # No OS images available for Windows2016
+    if not urns:
+        raise WindowsOsNotAvailableError()
```
[the remark this answers is not in the document]
Comment (dkulkarni-ms) [reply src/vm-repair/azext_vm_repair/repair_utils.py:514]: I am thinking to do this logic - 
If source vm os sku has gen2 or g2 then use specific 2022 datacenter gen 2 as sku
else will use specific 2022 datacenter gen1
 
That way it should work for both TVM and CVM.

File: src/vm-repair/azext_vm_repair/repair_utils.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dkulkarni-ms) [reply src/vm-repair/azext_vm_repair/repair_utils.py:472]: I have added to exit resource and clean up resource if the password is wrong.

File: src/vm-repair/azext_vm_repair/repair_utils.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (dkulkarni-ms) [reply src/vm-repair/azext_vm_repair/repair_utils.py:520]: Made necessary changes

**Pull request closed without merging**